src/query/api/v1/options/handler.go

```diff
@@ -71,7 +71,9 @@ type CustomHandler interface {
 	// Methods is the list of http methods this handler services.
 	Methods() []string
 	// Handler is the custom handler itself.
-	Handler(handlerOptions HandlerOptions) (http.Handler, error)
+	// prev is optional argument for getting already registered handler for the same route.
+	// If there is nothing to override, prev will be nil.
+	Handler(handlerOptions HandlerOptions, prev http.Handler) (http.Handler, error)
```

Comment:
This pattern seems like it may be a little fragile, mind if we catch up offline to discuss how it'll be used?